src/main/java/org/skife/jdbi/v2/unstable/BindIn.java

```diff
@@ -85,6 +86,42 @@ public void apply(SQLStatement q) throws SQLException
                 }
             };
         }
+
+		private static Collection<?> tryToCast(Object arg) {
+			if (arg instanceof Collection) {
+				return (Collection<?>) arg;
+			}
+			if (arg instanceof Object[]) {
```

Comment:
I'd probably like to see this the other way around: convert collections to an array and bind from the array. You can use `java.lang.reflect.Array.getLength(array)` and `Array.get(array, index)` and then you don't have to have all this code testing for different array types.

Also, there's a checkstyle error failing the build, because your code editor is using tabs instead of spaces for indentation. This project uses spaces.